dev-tools/mage/dockerbuilder.go

```diff
@@ -141,13 +144,42 @@ func (b *dockerBuilder) dockerSave(tag string) error {
 	// Save the container as artifact
 	outputFile := b.OutputFile
 	if outputFile == "" {
-		outputTar, err := b.Expand(defaultBinaryName + ".docker.tar")
+		outputTar, err := b.Expand(defaultBinaryName + ".docker.gz")
 		if err != nil {
 			return err
 		}
 		outputFile = filepath.Join(distributionsDir, outputTar)
 	}
-	if err := sh.Run("docker", "save", "-o", outputFile, tag); err != nil {
+	cmd := exec.Command("docker", "save", tag)
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
```

Comment:
Also connect `cmd.Stderr` to `os.Stderr`. Otherwise we will not get any errors. See documentation on `cmd.Cmd.StdErr`.

With piping contents and blocking on `Wait` it's might be a idea to read Stdout and Stderr via go-routines (at times the exec package is a little tricky to use).

Reply:
Stderr collected now and used as error message if command fails, an error would be printed like this now:
```
Error: failed building heartbeat type=docker for platform=linux/amd64: failed to save docker as artifact: exit status 1: Error response from daemon: reference does not exist
```